packages/cozy-realtime/src/index.spec.js

```diff
@@ -0,0 +1,239 @@
+    })
+
+    it('should launch handler when document is updated', async done => {
+      await realtime.onUpdate({ type }, doc => {
+        expect(doc).toEqual(fakeDoc)
+        done()
+      })
+
+      cozyStack.emitMessage(type, fakeDoc, 'UPDATED')
+    })
+
+    it('should launch handler when document with id is updated', async done => {
+      await realtime.onUpdate({ type, id: fakeDoc._id }, doc => {
+        expect(doc).toEqual(fakeDoc)
+        done()
+      })
+
+      cozyStack.emitMessage(type, fakeDoc, 'UPDATED', fakeDoc._id)
+    })
+
+    it('should launch handler when document is deleted', async done => {
+      await realtime.onDelete({ type }, doc => {
+        expect(doc).toEqual(fakeDoc)
+        done()
+      })
+
+      cozyStack.emitMessage(type, fakeDoc, 'DELETED')
+    })
+
+    it('should launch handler when document with id is deleted', async done => {
+      await realtime.onDelete({ type, id: fakeDoc._id }, doc => {
+        expect(doc).toEqual(fakeDoc)
+        done()
+      })
+
+      cozyStack.emitMessage(type, fakeDoc, 'DELETED', fakeDoc._id)
+    })
+
+    it('should relauch socket subscribe after an error', async () => {
+      const handler = jest.fn()
+      await realtime.onCreate({ type }, handler)
+      realtime._retryDelay = 100
+
+      expect(realtime._socket.isOpen()).toBe(true)
+      cozyStack.simulate('error')
+      expect(realtime._socket.isOpen()).toBe(false)
+      cozyStack.emitMessage(type, fakeDoc, 'CREATED')
+      expect(handler.mock.calls.length).toBe(0)
+
+      await pause(200)
+      expect(realtime._socket.isOpen()).toBe(true)
+      cozyStack.emitMessage(type, fakeDoc, 'CREATED')
+      expect(handler.mock.calls.length).toBe(1)
+    })
+  })
+
+  describe('unsubscribe', () => {
+    let handlerCreate, handlerUpdate, handlerDelete
+
+    beforeEach(async () => {
+      handlerCreate = jest.fn()
+      handlerUpdate = jest.fn()
+      handlerDelete = jest.fn()
+      await realtime.onCreate({ type }, handlerCreate)
+      await realtime.onUpdate({ type }, handlerUpdate)
+      await realtime.onDelete({ type }, handlerDelete)
+      await realtime.onCreate({ type: 'io.cozy.accounts' }, handlerCreate)
+    })
+
+    afterEach(() => {
+      realtime.unsubscribeAll()
+    })
+
+    it('should unsubscribe handlerCreate with type, eventName and handler', () => {
+      expect(realtime._socket.isOpen()).toBe(true)
+      expect(realtime._numberOfHandlers).toBe(4)
+      realtime.unsubscribe({ type, eventName: EVENT_CREATED }, handlerCreate)
+      expect(realtime._numberOfHandlers).toBe(3)
+      expect(realtime._socket.isOpen()).toBe(true)
+    })
+
+    it('should unsubscribe handlerCreate with type and eventName', () => {
+      expect(realtime._socket.isOpen()).toBe(true)
+      expect(realtime._numberOfHandlers).toBe(4)
+      realtime.unsubscribe({ type, eventName: EVENT_CREATED })
+      expect(realtime._numberOfHandlers).toBe(3)
```

Comment:
The `expect(realtime._numberOfHandlers)` is repeated several times, it should be done only once, in its own test.

Here it's written `should unsubscribe handlerCreate with type and eventName` but the test also tests the numberOfHandlers thing. Should the numberOfHandlers thing change, there would be several places to change.